### PR TITLE
poc - ws langs in nextjs

### PR DESCRIPTION
**Pull request by eagerterrier**
a potential solution for WS languages page

TODO: 
- ATI tracking for BBC_DEFAULT and hook it into all `ws` service models
- shipit

### CODE REVIEW

File: ws-nextjs-app/pages/ws/languages.page.tsx [hunk 1]
```diff
@@ -0,0 +1,529 @@
+import { GetServerSideProps } from 'next';
```
Comment (amoore108) [inline ws-nextjs-app/pages/ws/languages.page.tsx]: We'll need to import 'React' here

File: ws-nextjs-app/pages/ws/languages.page.tsx [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+      <ChartbeatAnalytics title={pageTitle} />
+      <MetadataContainer
+        title={pageTitle}
+        lang={lang}
+        description={pageDescription}
+        openGraphType="website"
+        hasAmpPage={false}
+      />
+      <style>{morphCSS1}</style>
+      <style>{morphCSS2}</style>
```
Comment (amoore108) [inline ws-nextjs-app/pages/ws/languages.page.tsx]: I'd wrap these in `Head` from `next/head`, like:
```      
<Head>
        <style>{morphCSS1}</style>
        <style>{morphCSS2}</style>
</Head>
```

Basically the same as `react-helmet`, but Nexts native implementation.

This should stop the hydration warnings.

File: ws-nextjs-app/pages/ws/languages.page.tsx [hunk 3]
```diff
@@ -0,0 +1,529 @@
...
+      <main>
+        <div id="page" className="atlas-languages-page">
+          <div className="container c-open" role="main">
+            <div className="column-clearfix">
+              <div className="column--primary">
+                <h1 className="story-body__h1">
+                  Get the news in your language
+                </h1>
+                <p className="language-switcher">
+                  <a
```
Comment (amoore108) [inline ws-nextjs-app/pages/ws/languages.page.tsx]: Can this be a `button` with `type=button` rather than an anchor if we're using it like a button?

**Review by amoore108 — CHANGES_REQUESTED** [review]

File: ws-nextjs-app/pages/ws/languages.page.tsx (hunk 3 above)
Comment (eagerterrier) [reply ws-nextjs-app/pages/ws/languages.page.tsx]: yeah, definitely _should be_

I just put what was there already.

Comment (eagerterrier) [reply ws-nextjs-app/pages/ws/languages.page.tsx]: I'll try and put a fix out

File: ws-nextjs-app/pages/ws/languages.page.tsx (hunk 2 above)
Comment (eagerterrier) [reply ws-nextjs-app/pages/ws/languages.page.tsx]: `Head` didn't work, so I put in `Helmet`

File: ws-nextjs-app/pages/ws/languages.page.tsx (hunk 1 above)
Comment (eagerterrier) [reply ws-nextjs-app/pages/ws/languages.page.tsx]: done

File: ws-nextjs-app/pages/ws/languages.page.tsx (hunk 2 above)
Comment (amoore108) [reply ws-nextjs-app/pages/ws/languages.page.tsx]: Hmm. Seemed fine when I tried it locally. Helmet is ok for now but if the Next one doesn't work we'll need to figure out why as we want to migrate off of using react-helmet.

**Comment by HarveyPeachey** [comment]
Is there any context around this work? Like a ticket or doc explaining what ws lang is?

**Comment by karinathomasbbc** [comment]
Playing devil's advocate here: could this page be an article or even a topic page?

**Review by shayneahchoon — APPROVED** [review]

**Comment by eagerterrier** [comment]
> Playing devil's advocate here: could this page be an article or even a topic page?

@karinathomasbbc  yeah - I wanted a static page, so we can do other similar things in nextjs, such as korean downloads, access to news etc. It also has a different ATI tracking ID.

**Comment by eagerterrier** [comment]
> Is there any context around this work? Like a ticket or doc explaining what ws lang is?

@HarveyPeachey good point. Not sure we do. https://www.bbc.co.uk/ws/languages

**Comment by amoore108** [comment]
If you wanted it to make it fully static, you could use `getStaticProps` instead of `getServerSideProps`, like:
```
export const getStaticProps: GetStaticProps = () => {
  return {
    props: {
      error: null,
      isAmp: false,
      isNextJs: true,
      page: null,
      pageData: null,
      pageType: STATIC_PAGE,
      pathname: '/ws/languages',
      service: 'ws',
      status: 200,
      timeOnServer: Date.now(), // TODO: check if needed?
    },
  };
};
```

Caveats are that we can't make use of logging here since the page is constructed once at build time, whereas `getServerSideProps` is ran on every request so we can access `res, req, headers` etc, so may not be an acceptable approach for us.

File: ws-nextjs-app/pages/ws/languages.page.tsx (hunk 2 above)
Comment (eagerterrier) [reply ws-nextjs-app/pages/ws/languages.page.tsx]: yeah - I pulled in using `next/document` rather than `next/head` - I've changed this so it now works 🎉

**Review by amoore108 — APPROVED** [review]
Looks good, worth trying it out and see how it behaves. Can come back and style up using the usual conventions / components if needed.

**Comment by eagerterrier** [comment]
> If you wanted it to make it fully static, you could use `getStaticProps` instead of `getServerSideProps`, like:

let's give it a go!

**Pull request merged**